content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md

```diff
@@ -0,0 +1,146 @@
+    SPRING_CLOUD_DATAFLOW_TASK_PLATFORM_CLOUDFOUNDRY_ACCOUNTS[foo]_CONNECTION_PASSWORD:
+    SPRING_CLOUD_DATAFLOW_TASK_PLATFORM_CLOUDFOUNDRY_ACCOUNTS[foo]_CONNECTION_SKIP_SSL_VALIDATION:
+    SPRING_CLOUD_DATAFLOW_TASK_PLATFORM_CLOUDFOUNDRY_ACCOUNTS[foo]_DEPLOYMENT_SERVICES:
+    SPRING_PROFILES_ACTIVE: cloud
+    JBP_CONFIG_SPRING_AUTO_RECONFIGURATION: '{enabled: false}'
+    SPRING_APPLICATION_JSON: '{"spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.brokers":"foo0.broker.foo,foo1.broker.foo,foo2.broker.foo","spring.cloud.dataflow.applicationProperties.stream.spring.spring.cloud.stream.kafka.binder.jaas.options.username":"test","spring.cloud.dataflow.applicationProperties.stream.spring.spring.cloud.stream.kafka.binder.jaas.options.password":"bestest","spring.cloud.dataflow.applicationProperties.stream.spring.spring.cloud.stream.kafka.binder.jaas.loginModule":"org.apache.kafka.common.security.plain.PlainLoginModule",
+    "spring.cloud.skipper.client.serverUri": "https://<SKIPPER_URI>/api"}'
+services:
+- mysql
+```
+
+With the above `manifest.yml`, SCDF should now be in the position to automatically propagated the required Kafka connection credentials to all the stream application deployments.
+
+```bash
+dataflow:>stream create fooz --definition "time | log"
+Created new stream 'fooz'
+
+dataflow:>stream deploy --name fooz
+Deployment request has been sent for stream 'fooz'
+```
+
+When the `time` and `log` applications are successfully deployed and started in Cloud Foundry, they should automatically connect to the external Kafka cluster.
+
+You can verify the connection credentials by accessing `time` or `log`'s `/configprops` actuator endpoints.
+Likewise, you will also see the connection credentials printed in the app logs.
+
+### Explicit Stream-level Kafka Connection Configuration
+
+Alternatively, if you intend to deploy only a particular stream with external Kafka connection credentials, you can do so when deploying a stream with explicit overrides.
+
+```bash
+dataflow:>stream create fooz --definition "time | log"
+Created new stream 'fooz'
+
+dataflow:>stream deploy --name fooz --properties "app.*.spring.cloud.stream.kafka.binder.brokers=foo0.broker.foo,foo1.broker.foo,foo2.broker.foo,app.*.spring.spring.cloud.stream.kafka.binder.jaas.options.username=test,app.*.spring.spring.cloud.stream.kafka.binder.jaas.options.password=besttest,app.*.spring.spring.cloud.stream.kafka.binder.jaas.loginModule=org.apache.kafka.common.security.plain.PlainLoginModule"
+Deployment request has been sent for stream 'fooz'
+```
+
+When the `time` and `log` applications are successfully deployed and started in Cloud Foundry, they should automatically connect to the external Kafka cluster.
+
+You can verify the connection credentials by accessing `time` or `log`'s `/configprops` actuator endpoints.
+Likewise, you will also see the connection credentials printed in the app logs.
+
```

Comment:
I think we should also document the CUPS workflow when using the SCDF file.

cf create-service p-dataflow standard data-flow -c '{"messaging-data-service": { "user-provided": {"bootstrap":"foo0.broker.foo","username":"test","password":"bestest"}}}'
 
And then reference properties as following in the individuals stream deployment properties:
 
${vcap.services.messaging-xxxxx.credentials.bootstrap} 
${vcap.services.messaging-xxxxx.credentials.username}
${vcap.services.messaging-xxxxx.credentials.password}

Reply:
I have not tried it, but I ported some of your output.